src/irep2/irep2_expr.cpp

```diff
@@ -388,7 +389,8 @@ void with2t::assert_consistency() const
 {
   if (is_array_type(source_value))
   {
-    assert(is_bv_type(update_field->type));
+    assert(
```

Comment:
Shall we use an assert or a log?

Reply:
This code will only run in the debug version, and based on the name of the function "assert_consistency()", I think assert is appropriate.